doc/release/upcoming_changes/25193.compatibility.rst

```diff
@@ -0,0 +1,6 @@
+The ``f2py.compile()`` helper has been removed due to a severe restriction in
```

Comment:
This makes it sound like it's purely due to Meson being worse than distutils, however the linked issue mentioned that it was marked as experimental, leaked memory and was anyway only a thin `subprocess.run` wrapper. So a rephrase here to mention those factors would be nice.

Reply:
Makes sense, reworded and squashed most of the history.